code/game/machinery/tesla_turret.dm

```diff
@@ -1,137 +1,619 @@
+		set_light(0)
+
+/obj/machinery/tesla_turret/proc/isLocked(mob/user)
+	if(ailock && issilicon(user))
+		to_chat(user, SPAN_NOTICE("There seems to be a firewall preventing you from accessing this device."))
+		return 1
+
+	if(locked && !issilicon(user))
+		to_chat(user, SPAN_NOTICE("Access denied."))
+		return 1
+
+	return 0
 
-	if(default_part_replacement(W, user))
+/obj/machinery/tesla_turret/attack_hand(mob/user)
+	if(!emagged || isLocked(user))
 		return
 
-	if(istype(W, /obj/item/card/id))
-		var/obj/item/card/id/C = W // New var to use ID-only vars.
-		if(!access_construction in C.access)
-			to_chat(user, "You do not have the required access to lock the turret.")
-			return
-		user.visible_message(
-							SPAN_NOTICE("[user] [locked ? "un" : ""]lock the [src.name]."),
-							SPAN_NOTICE("You [locked ? "un" : ""]lock the [src.name].")
-							)
-		locked = !locked
+	if(!anchored)
+		to_chat(usr, SPAN_NOTICE("\The [src] has to be secured first!"))
 		return
+	shock_net.ui_interact(user)
+
+/obj/machinery/tesla_turret/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = NANOUI_FOCUS)
+	var/data[0]
+	data["channel"] = shock_net.channel
+	data["access"] = 1
+	data["enabled"] = shock_net.enabled
+	data["is_lethal"] = 1
+	data["lethal"] = shock_net.lethal
+	var/settings[0]
+	settings[++settings.len] = list("category" = "Target Unauthorized Weapons", "setting" = "check_weapons", "value" = shock_net.check_weapons)
+	settings[++settings.len] = list("category" = "Target Unidentifiable Threats", "setting" = "check_records", "value" = shock_net.check_records)
+	settings[++settings.len] = list("category" = "Target Criminals", "setting" = "check_arrest", "value" = shock_net.check_arrest)
+	settings[++settings.len] = list("category" = "Target Unauthorized Colonists", "setting" = "check_access", "value" = shock_net.check_access)
+	settings[++settings.len] = list("category" = "Target All Synthetics", "setting" = "check_synth", "value" = shock_net.check_synth)
+	settings[++settings.len] = list("category" = "Target Fauna", "setting" = "check_anomalies", "value" = shock_net.check_anomalies)
+	settings[++settings.len] = list("category" = "Filter out Friendly Fauna", "setting" = "colony_allied_turret", "value" = shock_net.colony_allied_turret)
+	settings[++settings.len] = list("category" = "Toggle AI Access", "setting" = "ailock", "value" = shock_net.ailock)
+	data["settings"] = settings
+
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "tesla_turret_control.tmpl", "Tesla Turret Controls", 500, 300)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/machinery/tesla_turret/Topic(href, href_list)
+	if(..())
+		return 1
 
-	if(active)
-		to_chat(user, SPAN_NOTICE("You can't work with [src] while its running!"))
+	if(href_list["command"] && href_list["value"])
+		var/value = text2num(href_list["value"])
+		if(href_list["command"] == "enable")
```

Comment:
It would probably be better to do a `switch` statement here, instead of an if-else train.

Reply:
syntactically identical but it's fine either way